src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+        self.adjacency = (
+            np.zeros((num_nodes, num_nodes), dtype=int))
+        for n1 in self.graph.nodes:
+            for n2 in self.graph.nodes:
+                not_same = n1 != n2
+                is_conflict = self.is_edge_conflicting(n1, n2)
+                if not_same and is_conflict:
+                    self.graph.add_edge(n1, n2)
+                    self.adjacency[n1, n2] = 1
+
+    def rescale_adjacency(self):
+        """ Scale the adjacency matrix weights for QUBO solver. """
+        self.adjacency = np.triu(self.adjacency)
+        self.adjacency += self.adjacency.T - 2 * np.diag(
+            self.adjacency.diagonal())
+
+
+class SatelliteScheduleProblem(SchedulingProblem):
+    """
+    SatelliteScheduleProblem is a synthetic scheduling problem in which a
+    number of vehicles must be assigned to view as many requests in a
+    2-dimensional plane as possible. Each vehicle moves horizontally across
+    the plane, has minimum and maximum view angle, and has a maximum rotation
+    rate (i.e. the rate at which the vehicle can reorient vertically from one
+    target to the next).
+
+    The problem is represented as an infeasibility graph and can be solved by
+    finding the Maximum Independent Set.
+    """
+
+    def __init__(
+            self,
+            num_satellites: int = 6,
+            view_height: float = 0.25,
+            view_coords: Optional[np.ndarray] = None,
+            num_requests: int = 48,
+            requests: Optional[np.ndarray] = None,
+            turn_rate: float = 2,
+            solution_criteria: float = 0.99,
+            seed: int = 42,
+    ):
+        """ Create a SatelliteScheduleProblem.
+        Parameters
+        ----------
+        num_satellites : int, default = 6
+            The number of satellites to generate schedules for.
+        view_height : float, default = 0.25
+            The range from minimum to maximum viewable angle for each satellite.
+        view_coords : Optional[np.ndarray], default = None
+            The view coordinates (i.e. minimum viewable angle) for each
+            satellite in a numpy array. If None, view coordinates will be
+            evenly distributed across the viewable range.
+        num_requests : int, default = 48
+            The number of requests to generate.
+        turn_rate : float, default = 2
+            How quickly each satellite may reorient its view angle.
+        solution_criteria : float, default = 0.99
+            The target for a successful solution. The solver will stop
+            looking for a better schedule if the specified fraction of
+            requests is satisfied.
+        """
+        super(SatelliteScheduleProblem,
+              self).__init__(num_agents=num_satellites,
+                             num_tasks=num_requests,
+                             sat_cutoff=solution_criteria,
+                             seed=seed)
+        self.num_satellites = self.num_agents
```

Comment:
Is this link something that could go into a setter/getter?

Reply:
This is aliasing of the variables from the parent class to suit the nomenclature of the child class. A generic `SchedulingProblem` is now a `SatelliteSchedulingProblem`. So "agents" are now "satellites" and "tasks" are now "requests". Therefore, getter/setter is not suited here.